src/Engine/ProtoCore/AssociativeGraph.cs

```diff
@@ -69,11 +69,14 @@ public static int GetDirtyNodeCountAtGlobalScope(Executable exe)
             int dirtyNodes = 0;
             var graph = exe.instrStreamList[0].dependencyGraph;
             var graphNodes = graph.GetGraphNodesAtScope(Constants.kInvalidIndex, Constants.kGlobalScope);
-            foreach (AssociativeGraph.GraphNode graphNode in graphNodes)
+            if (graphNodes != null)
             {
-                if (graphNode.isDirty)
+                foreach (AssociativeGraph.GraphNode graphNode in graphNodes)
                 {
-                    ++dirtyNodes;
+                    if (graphNode.isDirty)
+                    {
+                        ++dirtyNodes;
+                    }
```

Comment:
The only place where `GetDirtyNodeCountAtGlobalScope()` is used is in `ApplyUpdate()` in LiveRunner.cs:
```
if (ProtoCore.AssociativeEngine.Utils.GetDirtyNodeCountAtGlobalScope(runnerCore.DSExecutable) > 0)
{
    ...
}
```
So it is not necessary to go through all graph nodes to answer a "Yes/No" question.